packages/desktop-gui/src/specs/specs-list.jsx

```diff
@@ -20,7 +20,39 @@ class SpecsList extends Component {
   render () {
     if (specsStore.isLoading) return <Loader color='#888' scale={0.5}/>
 
-    if (!specsStore.filter && !specsStore.specs.length) return this._empty()
+    const filteredSpecs = specsStore.getFilteredSpecs()
+
+    const hasSpecFilter = specsStore.filter
+    const numberOfShownSpecs = filteredSpecs.length
+    const hasNoSpecs = !hasSpecFilter && !numberOfShownSpecs
+
+    if (hasNoSpecs) {
+      return this._empty()
+    }
+
+    let runSpecsLabel = allSpecsSpec.displayName
+    let runButtonDisabled = false
+
+    if (hasSpecFilter) {
+      if (numberOfShownSpecs < 1) {
+        runSpecsLabel = 'No specs'
+        runButtonDisabled = true
+      } else {
+        const specLabel = numberOfShownSpecs === 1 ? 'spec' : 'specs'
+
+        runSpecsLabel = `Run ${numberOfShownSpecs} ${specLabel}`
+      }
+    }
+
+    const disabledMessage = (<span className={cs('all-tests', 'disabled')}>No specs</span>)
+
+    const runTestsMessage = (<a onClick={this._selectSpec.bind(this, allSpecsSpec)}
```

Comment:
I don't think there was any good reason why this was an anchor link instead of a button. Should probably make it a button cause `disabled` isn't an attribute that does anything inherently on `<a>` tags.